libsolidity/parsing/Parser.cpp

```diff
@@ -583,14 +584,62 @@ ASTPointer<UserDefinedTypeName> Parser::parseUserDefinedTypeName()
 	return nodeFactory.createNode<UserDefinedTypeName>(identifierPath);
 }
 
+ASTPointer<ElementaryTypeName> Parser::parseElementaryType()
+{
+	ASTNodeFactory nodeFactory(*this);
+	ASTPointer<ElementaryTypeName> type;
+	Token::Value token(Token::fromIdentifierOrKeyword(m_scanner->currentLiteral().substr(0, m_scanner->currentLiteral().find_first_of("1234567890"))));
+	if (Token::isElementaryTypeName(token))
+	{
+		string var, n, x, m;
+
+		size_t delimiter = m_scanner->currentLiteral().find_first_of("0123456789");
+		var = m_scanner->currentLiteral().substr(0, delimiter);
+
+		if (delimiter == string::npos)
+			type = ASTNodeFactory(*this).createNode<ElementaryTypeName>(token);
+		else if (delimiter != string::npos)
+		{
+			if (var == "bytes" || var == "int" || var == "uint")
+			{
+				n = m_scanner->currentLiteral().substr(delimiter);
+				cout << var << n << endl;
+				type = ASTNodeFactory(*this).createNode<ElementaryTypeName>(token, stoi(n));
+			}
+			else if (var == "real" || var == "ureal")
+			{
+				n = m_scanner->currentLiteral().substr(delimiter);
+				delimiter = n.find_first_of('x');
+				x = n.substr(delimiter);
+				delimiter = x.find_first_of("0123456789");
+				m = x.substr(delimiter);
+				if (delimiter == m.npos)
+					type = ASTNodeFactory(*this).createNode<ElementaryTypeName>(token, stoi(n), stoi(m));
+			}
+			else
+				fatalParserError(string("Elementary typename was not properly defined."));
+		}
+		
+		return type;
+	}
+	else 
+	{
+		fatalParserError(string("Expected elementary type name."));
+		return type;
+	}
+}
+
 ASTPointer<TypeName> Parser::parseTypeName(bool _allowVar)
 {
 	ASTNodeFactory nodeFactory(*this);
 	ASTPointer<TypeName> type;
 	Token::Value token = m_scanner->currentToken();
-	if (Token::isElementaryTypeName(token))
+
+	// because we no longer have explicit token names for types, it's now necessary to grab the current literal with no numbers
+	if (Token::isElementaryTypeName(Token::fromIdentifierOrKeyword(m_scanner->currentLiteral().substr(0, m_scanner->currentLiteral().find_first_of("1234567890")))))
```

Comment:
unless I understand otherwise, it is not possible to tell if it's an elementary type name purely from the token itself if the token has appended a number to it.